procedure/src/main/java/org/apache/iotdb/procedure/RootProcedureStack.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.procedure;
+
+import org.apache.iotdb.procedure.exception.ProcedureException;
+import org.apache.iotdb.service.rpc.thrift.ProcedureState;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class RootProcedureStack<Env> {
+  private static final Logger LOG = LoggerFactory.getLogger(RootProcedureStack.class);
+
+  private enum State {
+    RUNNING, // The Procedure is running or ready to run
+    FAILED, // The Procedure failed, waiting for the rollback executing
+    ROLLINGBACK, // The Procedure failed and the execution was rolledback
```

Comment:
rolling or rolled?

Reply:
It's a intermedia state for a rollback stack, means that there is at least one child proc is rolledback.